### PR TITLE
Plugin implementation

### PR DESCRIPTION
**Pull request by IgorPerikov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
-This plugin will provide a task for generating a Hollow consumer API, as described [here](http://hollow.how/getting-started/#consumer-api-generation).
+This plugin provides task for generating hollow consumer api, that is 
+described [here](http://hollow.how/getting-started/#consumer-api-generation)
+
+## Usage
+In order to use, add plugin dependency to your buildscript
+```
+buildscript {
+    repositories {
+        mavenLocal()
```
Comment (ghost) [inline README.md]: I don't believe `mavenLocal()` is necessary. We need it to work on the plugin ourselves, but once the plugin is released I think only `jcenter()` is required.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+        jcenter()
+    }
+    dependencies {
+        classpath "com.netflix.nebula:nebula-hollow-plugin:0.1"
+    }
+}
+```
+apply it
+```
+apply plugin: 'nebula.hollow-plugin'
```
Comment (ghost) [inline README.md]: We want this to be `apply plugin: 'nebula.hollow'`. I wonder what we need to change to make it work that way?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+contacts {
+    'perikov.igor@gmail.com' {
+        moniker 'Igor Perikov'
+        github 'IgorPerikov'
+    }
+}
+
+pluginBundle {
+    website = 'http://www.hollow.how'
+    vcsUrl = 'https://github.com/nebula-plugins/hollow-plugin'
```
Comment (ghost) [inline build.gradle]: Small typo: `/hollow-plugin` should be `/nebula-hollow-plugin`

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    description = 'This plugin provides a task for generating hollow consumer api.'
+    tags = ['hollow', 'nebula']
+
+    plugins {
+        hollowPlugin {
+            id = 'nebula.hollow-plugin'
```
Comment (ghost) [inline build.gradle]: Maybe `id = 'nebula.hollow'` is what we need to change it to?

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md]: I see your `TODO` further down regarding this. I believe there a TODO is fine. For this README we can write instructions assuming hollow plugin has been published.

**Review by ghost — CHANGES_REQUESTED** [review]
Looks like good stuff to me, @IgorPerikov. I only found a few changes I'd like to see. I'm also going to ask one of our gradle plugin experts to give this PR a review.

One general recommendation they made was to author the plugin in Java instead of Groovy. In their experience we've had performance problems with plugins written in Groovy. If you're interested in porting this to pure Java go ahead. This is a suggestion, not a requirement.

File: build.gradle (hunk 4 above)
Comment (IgorPerikov) [reply build.gradle]: yes, that's the 1st part of total work, 2nd is to rename `nebula.hollow-plugin.properties` to `nebula.hollow.properties`

File: README.md (hunk 2 above)
Comment (IgorPerikov) [reply README.md]: discussed here https://github.com/nebula-plugins/nebula-hollow-plugin/pull/1#discussion_r107259629

File: README.md (hunk 1 above)
Comment (IgorPerikov) [reply README.md]: ye, I agree with that

**Comment by IgorPerikov** [comment]
> One general recommendation they made was to author the plugin in Java instead of Groovy. In their experience we've had performance problems with plugins written in Groovy. If you're interested in porting this to pure Java go ahead. This is a suggestion, not a requirement.

Sure, I'll convert it to java7 implementation. Honestly, my first implementation was made with java, then I looked through most popular nebula-plugins and realised, that groovy might be more idiomatic for plugins :smile:

**Comment by IgorPerikov** [comment]
@toolbear 
https://github.com/nebula-plugins/nebula-hollow-plugin/pull/1/commits/898dd50756ecb11e436afbfaf845eaf6caf6a13e :

1. ports plugin source code from groovy to java
2. makes `apply plugin: 'nebula.hollow'` usage instead of `apply plugin: 'nebula.hollow-plugin'`
3. adds testcase, that checks that recursive scan works well
4. fixes `vcsUrl` typo and removes `mavenLocal()` section from readme

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
@IgorPerikov considering this is a brand new plugin that hasn't been announced, this is good to merge and we can iterate from there on other refinements.

Thanks for making this, BTW!

**Pull request merged**